### PR TITLE
Workflows and reorganization

### PR DESCRIPTION
**Pull request by csauve**
Sorry, this is a big one...

* Implement workflows, metabox content which displays the integrations between tools and file formats
* Unify page templates; it is now possible for any page to have any type of "feature" enhancement, such as workflows, tag tables, tag structure, and more without having to choose between fixed page types like tool, tag, or tagIndex.
* Reorganized rendering code to make things a bit clearer which components are related
* Reorganized page structure to put all halo 1 content under "/h1" and otherwise under "/general" so we can prepare for non-h1 content.

# Screenshots
tmp-workflows3
tmp-workflows4
tmp-workflows5
tmp-workflows2

### CODE REVIEW

File: src/content.js [hunk 1]
```diff
@@ -131,29 +132,35 @@ async function buildMetaIndex(contentDir, invaderDefsDir, baseUrl, packageVersio
...
   const findTagPageByName = (tagName) => {
-    const page = pages.find(page => page._slug == tagName && page.template == "tag");
+    const page = pages.find(page => page.tagName == tagName);
     if (!page) throw new Error(`Failed to find tag page for ${tagName}`);
     return page;
   };
 
   const findToolPageByName = (toolName) => {
-    const page = pages.find(page => page._slug == toolName.toLowerCase() && page.template == "tool");
+    const page = pages.find(page => page.toolName == toolName);
```
Comment (gbMichelle) [inline src/content.js:141]: Should we enforce case-correctness of toolnames? (I'm gravitating towards yes)

File: src/content/h1/sources/animation-data/readme.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  - to: ODX
+    for: JMO frames
+---
+The [HEK][] uses several file types as intermediate representations of animation data. These files provide a common target for exporters like [Bluestreak][] and [H2V Blender Animation Exporter][h2v-blender-animation-exporter], and can then be converted to [model_animations][] tags by [Tool][] for use in Halo.
+
+# JMA (Base)
+This is what a lot of animations will end up as. It is a default animation type that stores data necessary for movement, such as a [biped][] walking animation.
+
+# JMM (Base)
+...
```
Comment (gbMichelle) [inline src/content/h1/sources/animation-data/readme.md:20]: JMM is for movement, can we check to make sure this description is right?

File: src/content/h1/tags/model_collision_geometry/readme.md [hunk 3]
```diff
@@ -39,9 +39,9 @@ Pathfinding spheres are imported from the collision jms file of your object. The
 
 Pathfinding spheres can also be created automatically in some cases:
 
-* When an artist doesn't specify any pathfinding spheres will place one at the object's origin at half the size of the bounding sphere (which can be either too small or too big)
+* When an artist doesn't specify any pathfinding spheres, the game will assume one at the object's origin at half the size of the bounding sphere (which can be either too small or too big).
 * Mass points in [physics tags][physics] also count as pathfinding spheres. AI will actively avoid these.
-* [Units][unit] by default also have a pathfinding sphere around their feet at the size of the width of their physics pill.
+* [Units][unit] by default also have a pathfinding sphere around their feet with the same width as their physics pill.
```
Comment (gbMichelle) [inline src/content/h1/tags/model_collision_geometry/readme.md:44]: *Bipeds

Vehicles have pathfinding spheres based on their physics spheres. (These can be affected by the bounding sphere as AI will ignore pathfinding spheres outside of the bounding sphere.)

File: src/content/h1/tags/scenario_structure_bsp/readme.md [hunk 4]
```diff
@@ -64,7 +64,7 @@ When a JMS is compiled to BSP by [tool], connected convex faces with the materia
   </figcaption>
 </figure>
 
-Lightmaps are a special type of [bitmap][] referenced by the BSP which represents static lighting information. It is created using [Tool][] or [Sapien][] by the **[radiosity][]** process. A second set of UV coordinates is generated for the entire renderable BSP and a texture is rendered to apply levels of light to those surfaces.
+Lightmaps are a special type of [bitmap][] referenced by the BSP which represents static lighting information. It is created using [Tool][], [LM_Tool][], or [Sapien][] by the **[radiosity][]** process. A second set of UV coordinates is generated for the entire renderable BSP and a texture is rendered to apply levels of light to those surfaces.
```
Comment (gbMichelle) [inline src/content/h1/tags/scenario_structure_bsp/readme.md:67]: There exists Aether which can be used to generate lightmaps using 3dsmax

File: src/content/h1/tools/hek/readme.md [hunk 5]
```diff
@@ -1,10 +1,9 @@
 ---
 title: Halo Editing Kit (HEK)
-template: tool
+toolName: HEK
 info: |
-  * Download: http://hce.halomaps.org/index.cfm?fid=411
-  * Authors: Bungie, Gearbox Software
   * Release date: 2003
+  * [Download](http://hce.halomaps.org/index.cfm?fid=411)
```
Comment (gbMichelle) [inline src/content/h1/tools/hek/readme.md:6]: http://halocetutorials.tk has some useful mirrors for this tool

File: src/content/h1/tools/mek/readme.md [hunk 6]
```diff
@@ -1,17 +1,21 @@
 ---
 title: Mozz Editing Kit (MEK)
-template: tool
+toolName: MEK
 img: screenshot.png
 imgCaption: Screenshot of the MEK tool Mozzarilla, a replacement for [Guerilla][]
 info: |
-  [Development thread](https://opencarnage.net/index.php?/topic/7877-mozz-editing-kit-mek/)
+  * [Source](https://github.com/Sigmmma)
+  * [Development thread](https://opencarnage.net/index.php?/topic/7877-mozz-editing-kit-mek/)
```
Comment (gbMichelle) [inline src/content/h1/tools/mek/readme.md:8]: The threads never really get updated. Might consider putting something else in its place here

File: src/content.js (hunk 1 above)
Comment (csauve) [reply src/content.js:141]: I actually forgot to delete this function from earlier development. It's unused. When the metabox needs to link to an item involved in a workflow (e.g. `edit`, `from`, `to`, or `using`) or a similar item, it uses `resolvePage` with the page slug found in `tools.yml`.

File: src/content/h1/sources/animation-data/readme.md (hunk 2 above)
Comment (csauve) [reply src/content/h1/sources/animation-data/readme.md:20]: This is from looking at Bluestreak's source. JMM is:

* animation type == 0
* frame info type == 0
* world relative == false

The animation type enum is Base, Overlay, Replacement.

Comment (MosesofEgypt) [reply src/content/h1/sources/animation-data/readme.md:20]: Pool has help text on this
https://github.com/Sigmmma/hek_pool/blob/master/hek_pool/help_text.py#L82

File: src/content/h1/tools/mek/readme.md (hunk 6 above)
Comment (csauve) [reply src/content/h1/tools/mek/readme.md:8]: I'll put our discord as a support channel

File: src/content/h1/sources/animation-data/readme.md (hunk 2 above)
Comment (csauve) [reply src/content/h1/sources/animation-data/readme.md:20]: That's helpful info, thanks. I (or someone) can add that animations information in another pull request.

**Pull request merged**